pkg/cloudprovider/aws/apis/v1alpha1/provider_defaults.go

```diff
@@ -75,3 +77,24 @@ func (c *Constraints) defaultSecurityGroups(clusterName string) {
 	}
 	c.SecurityGroupSelector = map[string]string{fmt.Sprintf(ClusterDiscoveryTagKeyFormat, clusterName): "*"}
 }
+
+const (
+	DefaultMetadataOptionsHTTPEndpoint            = ec2.LaunchTemplateInstanceMetadataEndpointStateEnabled
+	DefaultMetadataOptionsHTTPPutResponseHopLimit = 2
+	DefaultMetadataOptionsHTTPTokens              = ec2.LaunchTemplateHttpTokensStateOptional
+)
+
+func (c *Constraints) defaultMetadataOptions() {
+	if c.MetadataOptions == nil {
+		c.MetadataOptions = &MetadataOptions{}
+	}
+	if c.MetadataOptions.HTTPEndpoint == "" {
+		c.MetadataOptions.HTTPEndpoint = DefaultMetadataOptionsHTTPEndpoint
+	}
+	if c.MetadataOptions.HTTPPutResponseHopLimit == 0 {
+		c.MetadataOptions.HTTPPutResponseHopLimit = DefaultMetadataOptionsHTTPPutResponseHopLimit
+	}
+	if c.MetadataOptions.HTTPTokens == "" {
```

Comment:
If it makes sense for this to potentially be nil, let's make it a ptr

Reply:
I'll update this too.